File: old/tags/2.2/cpp/DIS/SimulationAddress.cpp
#include <DIS/SimulationAddress.h> 

using namespace DIS;


SimulationAddress::SimulationAddress():
   _site(0), 
   _application(0)
{
}

SimulationAddress::~SimulationAddress()
{
}

unsigned short SimulationAddress::getSite() const
{
    return _site;
}

void SimulationAddress::setSite(unsigned short pX)
{
    _site = pX;
}

unsigned short SimulationAddress::getApplication() const
{
    return _application;
}

void SimulationAddress::setApplication(unsigned short pX)
{
    _application = pX;
}

void SimulationAddress::marshal(DataStream& dataStream) const
{
    dataStream << _site;
    dataStream << _application;
}

void SimulationAddress::unmarshal(DataStream& dataStream)
{
    dataStream >> _site;
    dataStream >> _application;
}


bool SimulationAddress::operator ==(const SimulationAddress& rhs) const
 {
     bool ivarsEqual = true;

     if( ! (_site == rhs._site) ) ivarsEqual = false;
     if( ! (_application == rhs._application) ) ivarsEqual = false;

    return ivarsEqual;
 }

int SimulationAddress::getMarshalledSize() const
{
   int marshalSize = 0;

   marshalSize = marshalSize + 2;  // _site
   marshalSize = marshalSize + 2;  // _application
    return marshalSize;
}

File: old/tags/2.2/java/edu/nps/moves/dis/SimulationAddress.java
package edu.nps.moves.dis;

import java.util.*;
import java.io.*;
import edu.nps.moves.jaxb.dis.*;

/**
 * Section 5.2.14.1. A Simulation Address  record shall consist of the Site Identification number and the Application Identification number.
 *
 * Copyright (c) 2007, MOVES Institute, Naval Postgraduate School. All rights reserved.
 * This work is licensed under the BSD open source license, available at https://www.movesinstitute.org/licenses/bsd.html
 *
 * @author DMcG
 */
public class SimulationAddress extends Object implements Serializable
{
   /** The site ID */
   protected int  site;

   /** The application ID */
   protected int  application;


/** Constructor */
 public SimulationAddress()
 {
 }

/** 
 * Constructor--takes a parallel jaxb object and returns an open-dis object 
 * 1.4_sed_bait_start */
 public SimulationAddress(edu.nps.moves.jaxb.dis.SimulationAddress x)
 {
     this.site = x.getSite();
     this.application = x.getApplication();
 }
/* 1.4_sed_bait_end */


/**
 * returns a jaxb object intialized from this object, given an empty jaxb object
 * 1.4_sed_bait_start **/
 public edu.nps.moves.jaxb.dis.SimulationAddress initializeJaxbObject(edu.nps.moves.jaxb.dis.SimulationAddress x)
 {
     ObjectFactory factory = new ObjectFactory();

     x.setSite( this.getSite() );
     x.setApplication( this.getApplication() );
   return x;
 }
/* 1.4_sed_bait_end */


public int getMarshalledSize()
{
   int marshalSize = 0; 

   marshalSize = marshalSize + 2;  // site
   marshalSize = marshalSize + 2;  // application

   return marshalSize;
}


public void setSite(int pSite)
{ site = pSite;
}

public int getSite()
{ return site; 
}

public void setApplication(int pApplication)
{ application = pApplication;
}

public int getApplication()
{ return application; 
}


public void marshal(DataOutputStream dos)
{
    try 
    {
       dos.writeShort( (short)site);
       dos.writeShort( (short)application);
    } // end try 
    catch(Exception e)
    { 
      System.out.println(e);}
    } // end of marshal method

public void unmarshal(DataInputStream dis)
{
    try 
    {
       site = dis.readShort();
       application = dis.readShort();
    } // end try 
   catch(Exception e)
    { 
      System.out.println(e); 
    }
 } // end of unmarshal method 


 /**
  * The equals method doesn't always work--mostly on on classes that consist only of primitives. Be careful.
  */
 public boolean equals(SimulationAddress rhs)
 {
     boolean ivarsEqual = true;

    if(rhs.getClass() != this.getClass())
        return false;

     if( ! (site == rhs.site)) ivarsEqual = false;
     if( ! (application == rhs.application)) ivarsEqual = false;

    return ivarsEqual;
 }
} // end of class
